src/wp-admin/includes/class-ftp.php

```diff
@@ -149,8 +149,11 @@ function __construct($port_mode=FALSE, $verb=FALSE, $le=FALSE) {
 	    $this->OS_local=FTP_OS_Unix;
 		$this->OS_remote=FTP_OS_Unix;
 		$this->features=array();
-		if(strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') $this->OS_local=FTP_OS_Windows;
-		elseif(strtoupper(substr(PHP_OS, 0, 3)) === 'MAC') $this->OS_local=FTP_OS_Mac;
+		if ( PHP_OS_FAMILY === 'Windows' ) {
+			$this->OS_local = FTP_OS_Windows;
+		} elseif ( PHP_OS_FAMILY === 'Darwin' ) {
+			$this->OS_local = FTP_OS_Mac;
+		}
```

Comment:
I'm not sure about this change. The `class-ftp.php` file is regarded as part of an [external dependency](https://www.phpclasses.org/package/1743-PHP-FTP-client-in-pure-PHP.html), so should generally not be touched unless to update the package.

Having said that, the external dependency is, AFAICS, no longer actively maintained... 🤷🏻‍♀️ 

Pro-tip: When in doubt about whether something is an external dependency or not, you can always check via the [`phpcs.xml.dist` file](https://github.com/WordPress/wordpress-develop/blob/f90c8bf1ae620b7568851f6d6e155223a8a6bf73/phpcs.xml.dist#L45-L78), as external dependencies don't have to comply with the WP Coding Standards.

Reply:
> Pro-tip: When in doubt about whether something is an external dependency or not, you can always check via the [phpcs.xml.dist file](https://github.com/WordPress/wordpress-develop/blob/f90c8bf1ae620b7568851f6d6e155223a8a6bf73/phpcs.xml.dist#L45-L78), as external dependencies don't have to comply with the WP Coding Standards.

TIL, thank you! 

I will remove this and push again.

Reply:
TIL, thank you!

I will remove this and push again.